com.unity.render-pipelines.core/Editor/AdditionalPropertiesPreferences.cs

```diff
@@ -0,0 +1,128 @@
+                    ).ToList();
+            }
+
+            if (s_IAdditionalPropertiesBoolFlagsHandlerTypes == null)
+            {
+                s_IAdditionalPropertiesBoolFlagsHandlerTypes = new List<(Type, Type)>();
+
+                var typeList = TypeCache.GetTypesDerivedFrom<IAdditionalPropertiesBoolFlagsHandler>()
+                    .Where(
+                        t => !t.IsAbstract && typeof(Editor).IsAssignableFrom(t)
+                    );
+
+                foreach (var editorType in typeList)
+                {
+                    var customEditorAttributes = editorType.GetCustomAttributes(typeof(CustomEditorForRenderPipelineAttribute), true);
+                    if (customEditorAttributes.Length != 0)
+                    {
+                        Type objectType = (Type)typeof(CustomEditor).GetField("m_InspectedType", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance).GetValue(customEditorAttributes[0]);
+                        s_IAdditionalPropertiesBoolFlagsHandlerTypes.Add((editorType, objectType));
+                    }
+                }
+            }
+        }
+
+        static bool showAllAdditionalProperties
+        {
+            get => s_ShowAllAdditionalProperties;
+            set
+            {
+                s_ShowAllAdditionalProperties = value;
+                EditorPrefs.SetBool(Keys.showAllAdditionalProperties, s_ShowAllAdditionalProperties);
+
+                ShowAllAdditionalProperties(showAllAdditionalProperties);
+            }
+        }
+
+        static class Keys
+        {
+            internal const string showAllAdditionalProperties = "General.ShowAllAdditionalProperties";
+        }
+
+        internal static void PreferenceGUI()
+        {
+            EditorGUI.BeginChangeCheck();
+            int newValue = EditorGUILayout.IntPopup(Styles.additionalPropertiesLabel, showAllAdditionalProperties ? 1 : 0, Styles.additionalPropertiesNames, Styles.additionalPropertiesValues);
+            if (EditorGUI.EndChangeCheck())
+            {
+                showAllAdditionalProperties = newValue == 1 ? true : false;
+            }
+        }
+
+        static void ShowAllAdditionalProperties(bool value)
+        {
+            // The way we do this here is to gather all types of either VolumeComponentEditor or IAdditionalPropertiesBoolFlagsHandler (for regular components)
+            // then we instantiate those classes in order to be able to call the relevant function to update the "ShowAdditionalProperties" flags.
+            // The instance on which we call is not important because in the end it will only change a global editor preference.
+            InitializeIfNeeded();
+
+            // Volume components
+            foreach (var editorType in s_VolumeComponentEditorTypes)
+            {
+                var editor = Activator.CreateInstance(editorType) as VolumeComponentEditor;
+                editor.InitAdditionalPropertiesPreference();
```

Comment:
I need an instance to toggle the pref to true or false so I can't just do it once from InitializeIfNeeded(). Otherwise I'd need to keep around the editors for ever and I'm not sure this is good.